pkg/metrics/nvme_test.go

```diff
@@ -0,0 +1,330 @@
+		}
+	}
+}
+
+func TestConvertHistogram(t *testing.T) {
+	tests := []struct {
+		name        string
+		histogram   Histogram
+		wantCount   uint64
+		wantBuckets map[float64]uint64
+	}{
+		{
+			name: "standard histogram with multiple bins",
+			histogram: Histogram{
+				BinCount: 3,
+				Bins: [64]HistogramBin{
+					{Lower: 0, Upper: 100, Count: 5},
+					{Lower: 100, Upper: 200, Count: 3},
+					{Lower: 200, Upper: 300, Count: 2},
+				},
+			},
+			wantCount: 10,
+			wantBuckets: map[float64]uint64{
+				100: 5,
+				200: 8,
+				300: 10,
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			gotCount, gotBuckets := convertHistogram(tt.histogram)
+
+			if gotCount != tt.wantCount {
+				t.Errorf("convertHistogram() count = %v, want %v", gotCount, tt.wantCount)
+			}
+
+			if !reflect.DeepEqual(gotBuckets, tt.wantBuckets) {
+				t.Errorf("convertHistogram() buckets = %v, want %v", gotBuckets, tt.wantBuckets)
+			}
+		})
+	}
+}
+
+func TestParseLogPage(t *testing.T) {
+	tests := []struct {
+		name    string
+		input   []byte
+		want    EBSMetrics
+		wantErr string
+	}{
+		{
+			name: "valid log page",
+			input: func() []byte {
+				metrics := EBSMetrics{
+					EBSMagic:              0x3C23B510,
+					ReadOps:               100,
+					WriteOps:              200,
+					ReadBytes:             1024,
+					WriteBytes:            2048,
+					TotalReadTime:         5000,
+					TotalWriteTime:        6000,
+					EBSIOPSExceeded:       10,
+					EBSThroughputExceeded: 20,
+				}
+				buf := new(bytes.Buffer)
+				if err := binary.Write(buf, binary.LittleEndian, metrics); err != nil {
+					t.Fatalf("failed to create test data: %v", err)
+				}
+				return buf.Bytes()
+			}(),
+			want: EBSMetrics{
+				EBSMagic:              0x3C23B510,
+				ReadOps:               100,
+				WriteOps:              200,
+				ReadBytes:             1024,
+				WriteBytes:            2048,
+				TotalReadTime:         5000,
+				TotalWriteTime:        6000,
+				EBSIOPSExceeded:       10,
+				EBSThroughputExceeded: 20,
+			},
+			wantErr: "",
+		},
+		{
+			name: "invalid magic number",
+			input: func() []byte {
+				metrics := EBSMetrics{
+					EBSMagic: 0x12345678,
+				}
+				buf := new(bytes.Buffer)
+				if err := binary.Write(buf, binary.LittleEndian, metrics); err != nil {
+					t.Fatalf("failed to create test data: %v", err)
+				}
+				return buf.Bytes()
+			}(),
+			want:    EBSMetrics{},
+			wantErr: "invalid EBS magic number: 12345678",
+		},
+		{
+			name:    "empty data",
+			input:   []byte{},
+			want:    EBSMetrics{},
+			wantErr: "failed to parse log page",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := parseLogPage(tt.input)
+
+			if tt.wantErr != "" {
+				if err == nil {
+					t.Errorf("parseLogPage() error = nil, wantErr %v", tt.wantErr)
+					return
+				}
+				if !strings.Contains(err.Error(), tt.wantErr) {
```

Comment:
Is there a reason we prefer string checking string errors instead of using `errors.Is(err.Error(), tt.wantErr)`? 

This causes brittle unit tests due to duplication because we have to update error strings in tests as well as pkg/metrics/nvme.go

Instead we can wrap errors in the nvme.go and re-use those errors in unit tests.

Reply:
Sure, wrapped errors in nvme.go so that we can re-use them in unit tests.